server/src/service.rs

```diff
@@ -125,15 +128,46 @@ impl Service {
             .name(oo7::dbus::api::Service::DESTINATION.as_deref().unwrap())?
             .build()
             .await?;
+        let object_server = connection.object_server();
         let service = Self {
+            collections: Mutex::new(Vec::new()),
             manager: Default::default(),
             connection: connection.clone(),
         };
+        let collections = service.init_collections().await;
 
-        connection
-            .object_server()
+        object_server
             .at(oo7::dbus::api::Service::PATH.as_deref().unwrap(), service)
             .await?;
+
+        for collection in collections {
+            object_server
+                .at(collection.path().clone(), collection)
+                .await?;
+        }
+
         Ok(())
     }
+
+    async fn init_collections(&self) -> Vec<Collection> {
+        let mut collections: Vec<Collection> = Vec::new();
+        // todo: create default collection
+
+        // create temporary session collection
+        let session = Collection::new(
```

Comment:
How do you identify this as a collection that never gets stored on disk? nothing tells so.

Reply:
The followings are the only two places mentioned about this that I could find.
[1] https://github.com/GNOME/gnome-keyring/blob/4132075144c7bb21b897570dd53b005ac38250aa/README#L6
[2] https://github.com/GNOME/libsecret/blob/e39b831776a932bd195571c1fa7d555702fd9284/libsecret/secret-collection.c#L81

Reply:
What I meant is that the code you wrote doesn't do anything for it to potentiallly work.  It needs work on the client::portal first as it doesn't have a way to create a new instance that never writes to disk.

Reply:
Ah yes I get it, thanks! But we still need this. Because, seahorse app won't work without this (not sure about Keyrack). During the launch of seahorse, it tries to retrieve the objectpath of session alias.

Reply:
I have pushed a commit that adds support for tmp keyring through  `oo7::portal::api::Keyring::temporary()`

Reply:
Very nice \o/, thanks!